session/auth/authprovider/authprovider.go

```diff
@@ -88,25 +88,24 @@ func (ap *authProvider) FetchToken(ctx context.Context, req *auth.FetchTokenRequ
 		ap.mu.Unlock()
 		// try GET first because Docker Hub does not support POST
```

Comment:
This is the comment for the Hub issue. If this is not true anymore we could switch the order that is potentially a clearer fix.

Reply:
@tonistiigi Docker Hub does support POST now. Hence, we can remove the comments.

Since not all token servers implement OAuth2, should we keep the existing logic or try OAuth2 first then fall back to basic auth?

Reply:
Updated the PR to the same logic as containerd by the commit 1b265c4.